bitbots_body_behavior/bitbots_body_behavior/actions/go_to_role_position.py

```diff
@@ -13,18 +13,19 @@ def __init__(self, blackboard, dsd, parameters=None):
         role_positions = self.blackboard.config['role_positions']
         try:
             if self.blackboard.blackboard.duty == 'goalie':
-                generalized_role_position = role_positions[self.blackboard.blackboard.duty]
+                generalized_role_position = role_positions['goalie']
             elif self.blackboard.blackboard.duty == 'offense':
                 kickoff_type = 'active' if self.blackboard.gamestate.has_kickoff() else 'passive'
                 if role_positions['pos_number'] == 0:
                     pos_number = 0
                 else:
                     # chose position randomly between left and right
                     pos_number = random.randint(1, 2)
-                generalized_role_position = role_positions[f'offense_{kickoff_type}_{pos_number}']
+                generalized_role_position = role_positions['offense'][f'{kickoff_type}_{pos_number}']
             else:
                 # players other than the goalie have multiple possible positions
-                generalized_role_position = role_positions[f'{self.blackboard.blackboard.duty}_{0}']
+                pos_number = role_positions['pos_number']
+                generalized_role_position = role_positions[f'defense_{pos_number}']
```

Comment:
I would not hardcode the defense role here, as the else case catches all roles (currently only defense exists, but who knows what we might add).